src/main/java/de/tum/in/www1/artemis/service/CompetencyProgressService.java

```diff
@@ -120,7 +120,7 @@ public void updateProgressByLearningObject(LearningObject learningObject, @NotNu
                 competencies = exerciseRepository.findByIdWithCompetencies(exercise.getId()).map(Exercise::getCompetencies).orElse(null);
             }
             else if (learningObject instanceof LectureUnit lectureUnit) {
-                competencies = lectureUnitRepository.findByIdWithCompetencies(lectureUnit.getId()).map(LectureUnit::getCompetencies).orElse(null);
+                competencies = lectureUnitRepository.findWithCompetenciesAndSlidesById(lectureUnit.getId()).map(LectureUnit::getCompetencies).orElse(null);
```

Comment:
Is it really necessary to fetch the slides to update the competency progress?

Reply:
I didn't want to create another repo method just for this case, but in the end it doesn't really matter and improves performance, so I added it now.